src/runtime/storageNG/storage-proxy.ts

```diff
@@ -34,4 +38,32 @@ export class StorageProxy<T extends CRDTTypeRecord> {
   getParticleView(): T['consumerType'] {
     return this.crdt.getParticleView()!;
   }
+
+  onMessage(message: ProxyMessage<T>): boolean {
+    // TODO: if InitializeProxy returns the id, we can assert that it is the same as the message id.
+    switch (message.type) {
+      case ProxyMessageType.ModelUpdate:
+        this.crdt.merge(message.model);
+        for (const handle of this.handles) {
+          if (handle.options.notifySync) {
+            handle.onSync();
+          }
+        }
+        break;
+      case ProxyMessageType.Operations:
+        for (const op of message.operations) {
+          this.crdt.applyOperation(op);
+        }
+        for (const handle of this.handles) {
+          if (handle.options.notifyUpdate) {
+            handle.onUpdate(message.operations);
+          }
+        }
+        break;
+      default:
```

Comment:
there's RequestSync too, which should be handled by *sending* the local model.

Reply:
added TODO